teku/src/main/java/tech/pegasys/teku/cli/subcommand/ValidatorClientCommand.java

```diff
@@ -94,9 +101,32 @@ public Integer call() {
     return 1;
   }
 
+  private void configureWithSpecFromBeaconNode(Eth2NetworkConfiguration.Builder builder) {
+    Spec spec = getSpecOrExit(validatorClientOptions.parseApiEndpoint());
+    builder.spec(spec);
+  }
+
+  private boolean isAutoDetectNetworkSpec() {
+    return AUTO_NETWORK_OPTION.equalsIgnoreCase(eth2NetworkOptions.getNetwork());
+  }
+
+  private void configureEth2Network(TekuConfiguration.Builder builder) {
+    if (isAutoDetectNetworkSpec()) {
+      builder.eth2NetworkConfig(this::configureWithSpecFromBeaconNode);
+    } else {
```

Comment:
Given that we will ultimately want to have a different default value for `--network` between the beacon node and validator client it may be worth removing the `Eth2NetworkOptions eth2NetworkOptions` mixin from `ValidatorClientCommand` and adding `--network` to `ValidatorOptions`.  Then the auto loading could be done in `ValidatorOptions.configure` rather than here which is probably ideal.

I'd probably just copy all the other options from `Eth2NetworkOptions` into a new `UnusedValidatorClientOptions` class and add that as a mixin.  That way they remain valid but ignored in case anyone has them on the command line.  We can remove that mixin in the future when we've given enough notice of them becoming invalid.

Reply:
sounds good ... I was thinking along similar lines initially but decided it might be best to go with an approach that minimizes change, albeit a bit hackish looking one

On a somewhat related note, what do you feel about `Eth2NetworkConfiguration` and that quite a few of the fields don't quite apply in the context of an external validator? Is it worth attempting to refactor this at this stage?

Reply:
> sounds good ... I was thinking along similar lines initially but decided it might be best to go with an approach that minimizes change, albeit a bit hackish looking one

Yeah I usually like doing things in small steps - I can just see that we'll have to do the bigger change so we can change the default anyway so may as well do it now.

> On a somewhat related note, what do you feel about `Eth2NetworkConfiguration` and that quite a few of the fields don't quite apply in the context of an external validator? Is it worth attempting to refactor this at this stage?

I think we need to keep the irrelevant options to preserve backwards compatibility for now, but if we copy them to a class only used by the validator client then we could actually mark them as `hidden = true` so they stop showing up in help.  Then we can list it as an upcoming breaking change in the readme for a couple of releases and remove them.  I doubt anyone is actually using them but better to avoid surprising people given it would mean their validator client doesn't startup anymore.